ontologytimemachine/utils/utils.py

```diff
@@ -80,28 +56,28 @@ def get_ontology_from_request(request):
             if v[0].decode('utf-8') == 'Host':
                 print('host found')
                 host = v[1].decode('utf-8')
+                path = request.path.decode('utf-8')
         ontology = 'https://' + host + request.path.decode('utf-8')
```

Comment:
why is this hardcoded to https???

Reply:
this hardcoded https seems really unsafe to me.
is there really no better way to get the scheme (protocol) - e.g. from the request object or by passing it as parameter?
keep in mind that this empty host seems to be a bug and could be fixed then this would break

Reply:
I agree, that it's not a nice solution, but so far I didn't find a better one. The scheme is not included anywhere in the https request, the only way to know that the current GET request is https is that it comes after a CONNECT and so far that the host is not defined. However, I don't think my code would fail in case the host gets fixed, because then it would fall into the else clause.

Reply:
Maybe the protocol should be read from the `request.protocol` parameter, cf. https://github.com/abhinavsingh/proxy.py/blob/develop/proxy/http/parser/parser.py#L67

Reply:
The value of the request.protocol is None in both cases